src/test/java/com/hpe/adm/octane/ideplugins/integrationtests/services/MetadataServiceITCase.java

```diff
@@ -36,5 +41,35 @@ public void testGetFormLayoutsForEntityType() {
         }
     }
 
+    @Test
+    public void testMethodForVisibleFields(){
+        Set<String> defaltFields = DefaultEntityFieldsUtil.getDefaultFields().get(Entity.USER_STORY);
+        List<FieldMetadata> fieldMetadata =(List) metadataService.getVisibleFields(Entity.USER_STORY);
+        List<String> fields = fieldMetadata.stream().map(FieldMetadata::getName).collect(Collectors.toList());
+        boolean flag = true;
+        for(String fieldName : defaltFields){
```

Comment:
@kdrtibor Iterating the set and then the list is O(n^2). If you revert the operation by iterating the list and checking if the element is in the set, it will be O(n)

Reply:
The reason why this is the order is because the default fields are always a subset of the requested fields. Not all returned fields must be default fields. Still to reduce the complexity i changed the list of returned field names to be implemented as a hash set.